model/repo_ref.go

```diff
@@ -102,3 +110,56 @@ func FindRepoRefByOwnerAndRepo(owner, repoName string) (*RepoRef, error) {
 		RepoRefRepoKey:  repoName,
 	}))
 }
+
+func (r *RepoRef) AddPermissions(creator *user.DBUser) error {
+	rm := evergreen.GetEnvironment().RoleManager()
+	catcher := grip.NewBasicCatcher()
+	if !r.Restricted {
+		catcher.Wrapf(rm.AddResourceToScope(evergreen.UnrestrictedProjectsScope, r.Id), "error adding repo project '%s' to list of unrestricted projects", r.Id)
+	}
+	catcher.Wrapf(rm.AddResourceToScope(evergreen.AllProjectsScope, r.Id), "error adding repo project '%s' to list of all projects", r.Id)
+	if catcher.HasErrors() {
+		return catcher.Resolve()
+	}
+
+	newScope := gimlet.Scope{
+		ID:          fmt.Sprintf("repo_%s", r.Id),
+		Resources:   []string{r.Id},
+		Name:        r.Id,
+		Type:        evergreen.ProjectResourceType,
+		ParentScope: evergreen.AllProjectsScope, // does this seem reasonable or will it result in duplicates?
```

Comment:
This is a question I guess I left in comments ^

Reply:
The part above isn't needed because adding the scope will add it to parents: https://github.com/evergreen-ci/gimlet/blob/master/rolemanager/role_manager.go#L189. However the parent here should be based on whether or not it's restricted I think